app/server/fulfilmentDateCalculatorReader.ts

```diff
@@ -0,0 +1,83 @@
+import Raven from "raven-js";
+import { getMainPlan, ProductDetail } from "../shared/productResponse";
+import { ProductTypes } from "../shared/productTypes";
+import { s3FilePromise } from "./awsIntegration";
+import { conf } from "./config";
+import { log } from "./log";
+
+interface RawFulfilmentDateCalculatorDates {
+  [dayOfWeek: string]: {
+    deliveryAddressChangeEffectiveDate?: string;
+    // there are a number of other fields, but we don't need them
+  };
+}
+
+const getDeliveryAddressChangeEffectiveDateForToday = async (
+  filenameProductPart: string,
+  daysOfWeek: string[]
+) => {
+  const datesByDayOfWeek = await s3FilePromise<
+    RawFulfilmentDateCalculatorDates
+  >(
+    `fulfilment-date-calculator-${conf.STAGE.toLowerCase()}`,
+    `${filenameProductPart}/${new Date()
+      .toISOString()
+      .substr(0, 10)}_${filenameProductPart}.json`,
+    ...daysOfWeek
+  );
+  return (
+    datesByDayOfWeek &&
+    daysOfWeek.reduce((earliestAcc, dayOfWeek) => {
+      const deliveryAddressChangeEffectiveDateStr =
+        datesByDayOfWeek[dayOfWeek].deliveryAddressChangeEffectiveDate;
+      if (
+        !earliestAcc ||
+        !deliveryAddressChangeEffectiveDateStr ||
+        new Date(earliestAcc) > new Date(deliveryAddressChangeEffectiveDateStr)
+      ) {
+        return deliveryAddressChangeEffectiveDateStr || "";
+      } else {
+        return earliestAcc;
+      }
```

Comment:
do you need the else here? could you just have:

`return earliestAcc;`

instead as that will only be called if the above if condition is not met

Reply:
done - https://github.com/guardian/manage-frontend/pull/338/commits/4f6df64f00b10e50b08ad35e711fc95959cfd3d0